src/syzygy/syzygy.cc

```diff
@@ -0,0 +1,1673 @@
+      uint16_t(*mapIdx)[2][2] =
+          be->hasPawns ? &PAWN(be)->dtmMapIdx[0] : &PIECE(be)->dtmMapIdx;
+      for (int t = 0; t < num; t++) {
+        for (int i = 0; i < 2; i++) {
+          mapIdx[t][0][i] = reinterpret_cast<uint16_t*>(data) + 1 - map;
+          data += 2 + 2 * read_le_u16(data);
+        }
+        if (split) {
+          for (int i = 0; i < 2; i++) {
+            mapIdx[t][1][i] = reinterpret_cast<uint16_t*>(data) + 1 - map;
+            data += 2 + 2 * read_le_u16(data);
+          }
+        }
+      }
+    }
+
+    if (type == DTZ) {
+      void* map = data;
+      *(be->hasPawns ? &PAWN(be)->dtzMap : &PIECE(be)->dtzMap) = map;
+      uint16_t(*mapIdx)[4] =
+          be->hasPawns ? &PAWN(be)->dtzMapIdx[0] : &PIECE(be)->dtzMapIdx;
+      uint8_t* flags =
+          be->hasPawns ? &PAWN(be)->dtzFlags[0] : &PIECE(be)->dtzFlags;
+      for (int t = 0; t < num; t++) {
+        if (flags[t] & 2) {
+          if (!(flags[t] & 16)) {
+            for (int i = 0; i < 4; i++) {
+              mapIdx[t][i] = data + 1 - static_cast<uint8_t*>(map);
+              data += 1 + data[0];
+            }
+          } else {
+            data += reinterpret_cast<uintptr_t>(data) & 0x01;
+            for (int i = 0; i < 4; i++) {
+              mapIdx[t][i] = reinterpret_cast<uint16_t*>(data) + 1 -
+                             static_cast<uint16_t*>(map);
+              data += 2 + 2 * read_le_u16(data);
+            }
+          }
+        }
+      }
+      data += reinterpret_cast<uintptr_t>(data) & 0x01;
+    }
+
+    for (int t = 0; t < num; t++) {
+      ei[t].precomp->indexTable = data;
+      data += size[t][0][0];
+      if (split) {
+        ei[num + t].precomp->indexTable = data;
+        data += size[t][1][0];
+      }
+    }
+
+    for (int t = 0; t < num; t++) {
+      ei[t].precomp->sizeTable = reinterpret_cast<uint16_t*>(data);
+      data += size[t][0][1];
+      if (split) {
+        ei[num + t].precomp->sizeTable = reinterpret_cast<uint16_t*>(data);
+        data += size[t][1][1];
+      }
+    }
+
+    for (int t = 0; t < num; t++) {
+      data = reinterpret_cast<uint8_t*>(
+          (reinterpret_cast<uintptr_t>(data) + 0x3f) & ~0x3f);
+      ei[t].precomp->data = data;
+      data += size[t][0][2];
+      if (split) {
+        data = reinterpret_cast<uint8_t*>(
+            (reinterpret_cast<uintptr_t>(data) + 0x3f) & ~0x3f);
+        ei[num + t].precomp->data = data;
+        data += size[t][1][2];
+      }
+    }
+
+    if (type == DTM && be->hasPawns) {
+      PAWN(be)->dtmSwitched =
+          calc_key_from_pieces(ei[0].pieces, be->num) != be->key;
+    }
+
+    return true;
+  }
+
+  int probe_table(const ChessBoard& pos, int s, int* success, const int type) {
+    // Obtain the position's material-signature key
+    Key key = calc_key_from_position(pos);
+
+    // Test for KvK
+    if (type == WDL && (pos.ours() + pos.theirs()).count() == 2) return 0;
```

Comment:
I think this is better as `if (type == WDL && pos.ours() == pos.our_king() && pos.theirs() == pos.their_king()) return 0;`